src/ucp/core/ucp_context.h

```diff
@@ -335,36 +335,44 @@ typedef struct ucp_tl_iface_atomic_flags {
                              UCS_BIT(UCT_ATOMIC_OP_CSWAP))
 
 
+/*
+ * Define UCP active message handler helper macro.
+ */
+#define _UCP_DEFINE_AM(_features, _id, _cb, _tracer, _flags, _proxy) \
+    ucp_am_handler_t ucp_am_handler_##_id  = { \
+        .features = _features, \
+        .cb       = _cb, \
+        .tracer   = _tracer, \
+        .flags    = _flags, \
+        .proxy_cb = _proxy \
+    }
+
+
 /*
  * Define UCP active message handler.
  */
 #define UCP_DEFINE_AM(_features, _id, _cb, _tracer, _flags) \
-    UCS_STATIC_INIT { \
-        ucp_am_handlers[_id].features = _features; \
-        ucp_am_handlers[_id].cb       = _cb; \
-        ucp_am_handlers[_id].tracer   = _tracer; \
-        ucp_am_handlers[_id].flags    = _flags; \
-    }
+    _UCP_DEFINE_AM(_features, _id, _cb, _tracer, _flags, NULL)
 
 
 /**
- * Defines a proxy handler which counts received messages on ucp_worker_iface_t
- * context. It's used to determine if there is activity on a transport interface.
+ * Defines UCP active message handler with proxy handler which counts received
```

Comment:
minor comment is: now, when we have UCP AM API, using `*UCP_AM*` for managing UCT AM handlers looks confusing. Maybe worth renaming since you already changing that

Reply:
@yosefe what do you think?

Reply:
no, it's just about the naming

Reply:
I see, any ideas about non-confusing name?

Reply:
Pls ignore, looks like we have too many such namings in the code. Does not make sense to handle this commen tin this pr

Reply:
maybe in next PR rename UCP_DEFINE_AM_xx to UCP_DEFINE_UCT_AM_xx

Reply:
yep, will do